src/Client/ClientPayloadFactory.php

```diff
@@ -72,12 +77,12 @@ public function createForAvailablePaymentMethods(
     public function createForPaymentDetails(
         array $receivedPayload,
         ?ShopperReferenceInterface $shopperReference = null,
-    ): array {
+    ): PaymentDetailsRequest {
```

Comment:
I wonder if it would be better to leave the array as the return type here and only use specific request classes in the client 🤔